src/braket/default_simulator/openqasm/circuit.py

```diff
@@ -63,28 +64,47 @@ def num_qubits(self) -> int:
 
     @property
     def basis_rotation_instructions(self):
-        """
-        This function assumes all observables are commuting.
-        """
-
         basis_rotation_instructions = []
-        measured_qubits = set()
+        observable_map = {}
+
+        def process_observable(observable):
+            if isinstance(observable, Identity):
+                return
+            measured_qubits = tuple(observable.measured_qubits)
+            for qubit in measured_qubits:
+                for target, previously_measured in observable_map.items():
+                    if qubit in target:
+                        conflicts = False
+                        if not (
+                            target == measured_qubits
+                            and type(previously_measured) == type(observable)
```

Comment:
This only works now because the only predefined observables are 1q Paulis; it might be worth future-proofing for if we get predefined multi-qubit observables.

Reply:
Moved around the logic a bit to make it cleaner. I believe it should be easier to follow and should be general for larger target standard observables now